game/scripts/vscripts/units/stopfightingyourself.lua

```diff
@@ -0,0 +1,194 @@
+require('libraries/garbagecollection')
+
+-- Entry Function
+function Spawn(entityKeyValues)
+  local disabled = false
+  if disabled then
+    return
+  end
+
+  print("Starting AI for " .. thisEntity:GetUnitName() .. " " .. thisEntity:GetEntityIndex())
+  thisEntity:SetContextThink('StopFightingYourselfThink', partial(Think, thisEntity), 1)
+
+  ABILITY_dupe_heroes = thisEntity:FindAbilityByName('boss_stopfightingyourself_dupe_heroes')
+
+  thisEntity.GC = GarbageCollection:Add(thisEntity.illusions, 10)
+end
+
+function Think(state, target)
+  -- NOTE: I'm thinking too long
+  -- BUG: Not attacking
+  -- BUG: illusions do not get deleted correctly
+  if not thisEntity:IsAlive() then
+    GarbageCollection:Remove(thisEntity.GC)
+    if thisEntity.illusions then
+      for i, illusion in ipairs(thisEntity.illusions) do
+        illusion:ForceKill(false)
+        if not illusion:IsNull() then
+          illusion:RemoveSelf()
+        end
+      end
+      thisEntity.illusions = nil
+    end
+    return 0
+  end
+
+  -- Leash
+  if not GLOBAL_origin then
+    GLOBAL_origin = thisEntity:GetAbsOrigin()
+  else
+    local distance = (GLOBAL_origin - thisEntity:GetAbsOrigin()):Length()
+    if distance > 1000 then
+      ExecuteOrderFromTable({
+        UnitIndex = thisEntity:entindex(),
+        OrderType = DOTA_UNIT_ORDER_MOVE_TO_POSITION,
+        Position = GLOBAL_origin, --Optional.  Only used when targeting the ground
+        Queue = 0 --Optional.  Used for queueing up abilities
+      })
+      return 5
+    end
+  end
+
+  if thisEntity:IsIdle() and IsHeroInRange(thisEntity:GetAbsOrigin(), 900) then
+    UseRandomItem()
```

Comment:
I've only ever seen people assign queue to a number, if you're having problems with only the last thing you run working or only the one you put a 0 in working, i would change this to instead either queue up sequentially or fall through conditional if it worked so that they don't stomp over each other.